pkgs/poetry/default.nix

```diff
@@ -0,0 +1,34 @@
+{ pkgs }:
+let
+  python = pkgs.python310Full;
+
+  myPoetry = pkgs.stdenv.mkDerivation {
+    name = "poetry";
+    version = "1.11.1";
+    src = ./.;
+
+    buildInputs = [ pkgs.makeWrapper ];
+
+    installPhase = ''
+      mkdir -p $out
+      mkdir -p $out/bin
+
+      ${python}/bin/python3 -m venv $out/env
+      touch $out/env/poetry_env # This allows poetry to recognize it
+                                # https://github.com/replit/poetry/blob/replit-1.1/poetry/utils/env.py#L885
+                                # invoking the workaround so that poetry
+                                # does not use its own venv for the project
+                                # env
+
+      mkdir -p $out/poetry.bundle
+      tar xvf poetry.bundle.tar.gz -C $out/poetry.bundle
```

Comment:
Can we convert this to be a fixed-output-derivation (`pkgs.fetchTarball`), so we don't need to check in the bundle?

Reply:
Happy to pair with you to try to do this!

Reply:
But then we'd have to publish the tarball somewhere?